addons/sale_timesheet/wizard/sale_make_invoice_advance.py

```diff
@@ -0,0 +1,53 @@
+from odoo import api, fields, models
+
+
+class SaleAdvancePaymentInv(models.TransientModel):
+    _inherit = "sale.advance.payment.inv"
+
+    @api.model
+    def _default_invoicing_timesheet_enabled(self):
+        product_ids = None
+        if self._context.get('active_id', False):
+            product_ids = self.env['sale.order'].browse(self._context.get('active_id')).order_line.mapped('product_id')
+
+        elif self._context.get('active_ids', []):
+            order_lines = self.env['sale.order'].browse(self._context.get('active_ids')).mapped('order_line')
+            product_ids = order_lines.mapped('product_id')
+
+        if product_ids:
+            return any(product.is_delivered_timesheet for product in product_ids)
+        return False
+
+    date_invoice_timesheet = fields.Date(
+        string='Date invoice timesheet',
+        default=fields.Date.today(),
+        required=False,
+        help="This field is used to invoice timesheets up to this date")
+    invoicing_timesheet_enabled = fields.Boolean(default=_default_invoicing_timesheet_enabled)
+
+    def create_invoices(self):
+        """ Override method from sale/wizard/sale_make_invoice_advance.py
+
+            When the user want to invoice the timesheets to the SO
+            up to a specific date then we need to recompute the
+            qty_to_invoice for each product_id in sale.order.line,
+            before creating the invoice.
+        """
+        sale_orders = self.env['sale.order'].browse(
```

Comment:
You can store the ``order_ids`` on the wizard, and set it in the ``default_get``. That way, you don't have to make all those ``self._context.get('active_ids', [])``, and by the way, if you create this wizard in a python code, you just have to set 'order_id' in the ``vals`` dictionary instead of playing with ``with_context`` and hoping for the best ;)